nrf_802154/serialization/src/nrf_802154_spinel_app.c

```diff
@@ -279,6 +279,49 @@ static nrf_802154_ser_err_t capabilities_await(uint32_t                    timeo
     return error;
 }
 
+/**
+ * @brief Wait with timeout for time property to be received.
+ *
+ * @param[in]  timeout  Timeout in us.
+ * @param[out] p_time   Pointer to the time variable which needs to be populated.
+ *
+ * @returns  zero on success or negative error value on failure.
+ *
+ */
+static nrf_802154_ser_err_t time_await(uint32_t   timeout,
+                                       uint32_t * p_time)
+{
+    nrf_802154_ser_err_t              res;
+    nrf_802154_spinel_notify_buff_t * p_notify_data = NULL;
+
+    SERIALIZATION_ERROR_INIT(error);
+
+    p_notify_data = nrf_802154_spinel_response_notifier_property_await(
+        timeout);
+
+    SERIALIZATION_ERROR_IF(p_notify_data == NULL,
+                           NRF_802154_SERIALIZATION_ERROR_RESPONSE_TIMEOUT,
+                           error,
+                           bail);
+
+    res = nrf_802154_spinel_decode_prop_nrf_802154_time_get_ret(p_notify_data->data,
+                                                                p_notify_data->data_len,
+                                                                p_time);
+
+    SERIALIZATION_ERROR_CHECK(res, error, bail);
+
+    NRF_802154_SPINEL_LOG_BANNER_RESPONSE();
+    NRF_802154_SPINEL_LOG_VAR_NAMED("%ul", *p_time, "Time");
```

Comment:
Why 'l' at the end of the format string?

Reply:
Isn't it for `long`? `p_time` is the pointer to the `uint32_t`, which corresponds to the `long` one.

Reply:
So I guess it should be `%lu` in such a case? With the existing code, the `l` will be printed after the value.

Reply:
Maybe using PRIu32 for print formatting uint32_t will be better solution?